<launch>
    <?ignore
    <include file="$(find setting_tf2)/launch/tf2_setting.launch">
    </include>

    <!-- Dynamixel Workbench Controller -->
    <include file="$(find dynamixel_workbench_controllers)/launch/position_control.launch">
        <arg name="device_name"        		default="/dev/ttyU2D2"/>
        <arg name="baud_rate"   		default="4500000"/>
	<arg name="profile_velocity"		default="0"/>
	<arg name="profile_acceleration"	default="0"/>
	<arg name="position_p_gain"		default="1500"/>
	<arg name="position_i_gain"		default="300"/>
	<arg name="position_d_gain"		default="100"/>
    </include>

    <!-- ros_mscl -->
    <include file="$(find microstrain_inertial_driver)/launch/microstrain.launch">
          <arg name="name"                   default = "gx5" />
	  <arg name="port"                   default = "/dev/3dm-gx4-25" />
	  <arg name="aux_port"               default = "/dev/ttyACM1" />
	  <arg name="baudrate"               default = "115200" />
	  <arg name="debug"                  default = "false" />
	  <arg name="diagnostics"            default = "false" />
	  <arg name="imu_frame_id"           default = "sensor" />
	  <arg name="imu_data_rate"          default = "150" />
	  <arg name="filter_data_rate"       default = "0" />
	  <arg name="gnss1_frame_id"         default = "gnss1_antenna_wgs84" />
	  <arg name="gnss2_frame_id"         default = "gnss2_antenna_wgs84" />
	  <arg name="filter_frame_id"        default = "sensor_wgs84" />
	  <arg name="filter_child_frame_id"  default = "sensor" />
	  <arg name="nmea_frame_id"          default = "nmea" />
	  <arg name="use_enu_frame"          default = "false" />
    </include>

    <!-- t265 -->
    <include file="$(find realsense2_camera)/launch/rs_t265.launch">
    </include>
    ?>
    <!-- rosserial_arduino -->
    <node pkg="rosserial_arduino" type="serial_node.py" name="ros_arduino_PWMs" >
	<param name="port" value="/dev/arduino-dueP"/>
        <param name="baud" value="57600"/>
    </node>
    
    <node pkg="pwm_pca9685" type="pca9685_node" name="pwm_generator"/>

    <!-- T3_MAV_Controller -->
	<!--integration limits-->
	<param name="attitude_integ_limit"	value="10.0"/>
	<param name="altitude_integ_limit"	value="100.0"/>
	<param name="position_integ_limit"	value="10.0"/>

    	<!--attitude_rp gains-->
    	<param name="attitude_rp_P_gain"	value="4.5"/>
    	<param name="attitude_rp_I_gain"	value="0.5"/>
    	<param name="attitude_rp_D_gain"	value="0.7"/>

	<!--attitude_y gain-->
	<param name="attitude_y_P_gain"		value="2.0"/>
	<!--<param name="attitude_y_I_gain"		value="0"/>-->
	<param name="attitude_y_D_gain"		value="0.2"/>

	<!--Altitude gains-->
	<param name="altitude_P_gain"		value="20.0"/>
	<param name="altitude_I_gain"		value="5.0"/>
	<param name="altitude_D_gain"		value="15.0"/>

	<!--Position gains-->
	<param name="position_P_gain"		value="5.0"/>	
	<param name="position_I_gain"		value="0.1"/>	
	<param name="position_D_gain"		value="5.0"/>	

	<!--Center of mass-->
	<param name="x_center_of_mass"		value="0.0"/>	
	<param name="y_center_of_mass"		value="0.0"/>	
	<param name="z_center_of_mass"		value="-0.03"/>	


    <node name="t3_mav_controller_node"  pkg="t3_mav_controller" type="t3_mav_controller_node"  output="screen"/>

    <!-- Rosbag -->
    <node pkg="rosbag" type="record" name="record" output="screen" args="-o /home/pwyong/Desktop/catkin_ws/src/t3_mav_controller/bag/ -a" /><!-- /angle /desired_angle /Forces /t265_pos /desired_altitude"/> -->
</launch>
